kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -989,7 +995,26 @@ protected T waitUntilExists(long amount, TimeUnit timeUnit) throws InterruptedEx
 
   @Override
   public T waitUntilReady(long amount, TimeUnit timeUnit) throws InterruptedException {
-    return waitUntilExists(amount, timeUnit);
+
+    long timeoutInNanos = timeUnit.toNanos(amount);
+    long end = System.nanoTime() + timeoutInNanos;
+
+    while (System.nanoTime() < end) {
+      T item = fromServer().get();
+
+      if (Readiness.isReady((HasMetadata) item)) {
```

Comment:
This call may through an Exception if `waiting` makes no sense for the resource.

When used for a single resource it's ok, but if batch waiting delegates to this call, this may be problematic.